packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/endpoint_client/endpoint_global_fields_manager.dart

```diff
@@ -0,0 +1,110 @@
+import 'dart:collection';
+import 'dart:convert';
+import '../shared_prefs.dart';
+
+class EndpointGlobalFieldsManager {
```

Comment:
Can you add doc comments throughout? I know this is still a draft, but it makes reviewing it much easier to read a comment and get a quick understanding of what this class does.